gpgme: Fix build with python on macOS

kirelagin opened the pull request:
* Replace LD_LIBRARY_PATH with OS-specific name (e.g. DYLD_LIBRARY_PATH
  on macOS).
* Disable Python tests on macOS, because they use gpg, which fails due
  to a very long socket path (https://github.com/NixOS/nix/pull/1085).

The former should be fixed upstream ([patch](https://lists.gnupg.org/pipermail/gnupg-devel/2020-April/034591.html)). The latter is a Nix-specific issue,
but it can be worked-around upstream by making Python tests respect
--disable-gpg-test.

Fixes #70996.
Related NixOS/nix#1085.

<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

kirelagin committed:
pgpgme: Fix build with python on macOS

* Replace LD_LIBRARY_PATH with OS-specific name (e.g. DYLD_LIBRARY_PATH
  on macOS).
* Disable Python tests on macOS, because they use gpg, which fails due
  to a very long socket path (https://github.com/NixOS/nix/pull/1085).

The former should be fixed upstream. The latter is a Nix-specific issue,
but it can be worked-around upstream by making Python tests respect
--disable-gpg-test.

Mic92 commented on pkgs/development/libraries/gpgme/default.nix:
```diff
@@ -49,6 +49,15 @@ stdenv.mkDerivation rec {
+  # Apply only on Darwin to save rebuilds on Linux
+     + lib.optionalString stdenv.isDarwin ''
+    sed -i "s/LD_LIBRARY_PATH/@shlibpath_var@/" lang/python/tests/Makefile.in
+    sed -i "s/ac_subst_vars='/ac_subst_vars='shlibpath_var\n/" configure
```
It would be nice to also send this one upstream: https://dev.gnupg.org/

Mic92 replied:
cc @AndreHeinecke

kirelagin replied:
I sent it.

Mic92 replied:
Sounds good. Can you provide a link?

kirelagin replied:
They accept patches by email, and, for some reason, it hasn’t appeared in https://lists.gnupg.org/pipermail/gnupg-devel/2020-April/ yet.

kirelagin replied:
https://lists.gnupg.org/pipermail/gnupg-devel/2020-April/034591.html

Mic92 committed:
gpgme: convert macos sed patch scripts to patch files

This is less brittle and breaks loud if the code changes.
Also remove the /usr/bin/file patch. It is not really required
for the build to work, the generated warning is harmless.

Mic92 commented on pkgs/development/libraries/gpgme/default.nix:
```diff
@@ -33,7 +34,14 @@ stdenv.mkDerivation rec {
+      sha256 = "00d4sxq63601lzdp2ha1i8fvybh7dzih4531jh8bx07fab3sw65g";
+    })
+    # Disable python tests on Darwin as they use gpg (see configureFlags below)
+  ] ++ lib.optional stdenv.isDarwin ./disable-python-tests.patch;
```
This way it is easier to maintain and less likely to accidentally to break on updates.

Mic92 replied:
Please let me know if it still works.

kirelagin replied:
It works.
